### ISSUE TITLE
Compare meta data between old PeMS and Modern PeMS

### ISSUE THREAD

**Issue by mmmiah** [cagov/caldata-mdsa-caltrans-pems#497]
We need to compare meta data between old and modern pems to understand the consistency between these two platform. Lets grab one week data from INT_DIAGNOSTICS__DETECTOR_STATUS (modern PeMS ) and DETECTOR_STATUS (Old PeMS) and compare statistically and visually. Summarize the findings in this issue. It is also related #398

**Comment by mmmiah** [comment]
**Analysis Topic :** Spatial and Temporal Validation of Old and Moder PeMS Meta Data
**Goal :** Conduct the Spatial and Temporal Validation of old and Modern PeMS meta data

**Data Source:** Seven days daily level data was downloaded from old and modern PeMS where time windows vary from 2024-11-01 to 2024-11-07.  The details are shown in below table-

![image](https://github.com/user-attachments/assets/cb0a92c8-1bed-4a92-80fe-ec1de64eef5d)

**Data Processing/Methodology:** 

Step 1: Download data from old and Modern PeMS for same time windows and count them to compare
Step 2: Match these two data from based on following conditions-
                        Old PeMS [Timestamp]= Modern PeMS [Timestamp]
                        Old PeMS [Station Id]= Modern PeMS [Station Id]
                        Old PeMS [Detector Id]= Modern PeMS [Detector Id]]
                        Old PeMS [Freeway Id]= Modern PeMS [Freeway Id]
                        Old PeMS [Freeway Direction]= Modern PeMS [Freeway Direction]
                        Old PeMS [District Id]= Modern PeMS [District Id]
Step 3: Compare the Detector Count by Sample Date, and Caltrans District
Step 4: Re-match old and Modern PeMS data set with additional condition along with all previous condition-
                        Old PeMS [Status]= Modern PeMS [Status]
Step 5: Compare the Good and bad Status count between these two data frame by District

**Analysis and Results** : It was observed that the detector counts showed no difference between the old and modern PeMS datasets for the considered time windows, indicating a perfect temporal match between the two datasets.
![image](https://github.com/user-attachments/assets/62876e49-e9d4-4f53-83e3-3f95786c99e5)

We also analyzed the metadata count spatially by district and found no differences between the old and modern PeMS metadata, as shown in the table below-
![image](https://github.com/user-attachments/assets/48701a98-0f0e-45a3-adf6-c4251178f79a)
Finally, we compared the daily health status between the two data frames and found that 6,909 detectors had mismatched health statuses between the old and modern PeMS, as presented in the table below.
![image](https://github.com/user-attachments/assets/adf0b3c0-0066-427a-a894-7f22afc9b113)
This inconsistency accounts for only 2.07% of the seven-day data, indicating a minor discrepancy in health status between the two data frames that needs to be addressed.

The [attached]([modern_Pems_unmatched_status_data.csv](https://github.com/user-attachments/files/18129625/modern_Pems_unmatched_status_data.csv)) is the list of 6,909 rows from the old and modern PeMS datasets. This includes a column labeled status_diff, which highlights the locations and details of these inconsistencies.

**Findings** : 
1. It was found that old and Modern PeMS meta data have 100% matching in terms of spatial and Temporal detector count
2. However, we found that 2.07% deviation of detector status match between old and modern PeMS
Action Items: ken is working to solve the issue through #398

**Comment by mmmiah** [comment]
PPT for this Analysis : 
[Meta Data Comparision_PPT.pptx](https://github.com/user-attachments/files/18129720/Meta.Data.Comparision_PPT.pptx)
A ArcGIS Dashboard : https://d3giscoreage.dot.ca.gov/portal/apps/dashboards/5258ed24ffb2412f94e3dd9c5d9f671b

**Comment by mmmiah** [comment]
@kengodleskidot , Thank you for revising the detector status criteria. After re-running the script, I observed that the detector status mismatch decreased to 0.94% (3,128 data points) from 2.07% (6,909 data points) over a seven-day analysis window. It appears that most of the remaining inconsistencies are currently in D8. We have done our best to align with the old PeMS data. I will keep this issue open for a couple of days to allow for any comments before closing it.
![image](https://github.com/user-attachments/assets/6401125f-0a37-41c7-889b-d93efc63b355)
I have also attached the remaining updated inconsistency table for any future reference. 
[modern_Pems_unmatched_status_data.csv](https://github.com/user-attachments/files/18204177/modern_Pems_unmatched_status_data.csv)
